app/services/create_newsletter_signup.rb

```diff
@@ -0,0 +1,46 @@
+require 'sendgrid-ruby'
+include SendGrid
+
+class CreateNewsletterSignup
+  include ActiveModel::Validations
+
+  validates :newsletter_signup, presence: true
+
+  attr_accessor :newsletter_signup
+
+  def initialize(newsletter_signup:)
+    @newsletter_signup = newsletter_signup
```

Comment:
What is a newsletter signup? A hash? What if it’s missing the requisite keys?

Let’s use some simple Ruby Keyword Variables here on this service object.

Reply:
I added some error handling for cases where the keys were omitted from the form submission. I was unsure whether I should focus on using the SendGrid, ServiceResponse, or KeywordError object response for this. I might have gotten further in the weeds but it's doing the job for me locally.